Blackbaud.CustomFx.ConstituentPhotoAnalysis.Catalog/ConstituentPhotoAnalysis.BusinessProcess.cs

```diff
@@ -0,0 +1,179 @@
+using Blackbaud.AppFx.Server;
+using Blackbaud.AppFx.Server.AppCatalog;
+using Blackbaud.CustomFx.ConstituentPhotoAnalysis.SPWrapConstituentPhotoAnalysis;
+using Microsoft.Azure.CognitiveServices.Vision.Face;
+using Microsoft.Azure.CognitiveServices.Vision.Face.Models;
+using System.Data.SqlClient;
+using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Blackbaud.CustomFx.ConstituentPhotoAnalysis.Catalog
+{
+    public sealed class ConstituentPhotoAnalysisBusinessProcess : AppBusinessProcess
+    {
+
+        // You will need your own subscription key
+        private const string subscriptionKey = "<Replace with your key>";
+
+        //You must use the same region as you used to get your subscription
+        // keys. For example, if you got your subscription keys from westus,
+        // replace "westcentralus" with "westus".
+
+        // Free trial subscription keys are generated in the westcentralus
+        // region. If you use a free trial subscription key, you shouldn't
+        // need to change the region.
+        // Specify the Azure region
+        private const string faceEndpoint = "https://westcentralus.api.cognitive.microsoft.com";
+
+        private static readonly FaceAttributeType[] faceAttributes = {
+            FaceAttributeType.Age,
+            FaceAttributeType.Gender,
+            FaceAttributeType.Emotion,
+            FaceAttributeType.Smile
+        };
+
+        private static FaceClient faceClient;
+
+        public override AppBusinessProcessResult StartBusinessProcess()
+        {
+            AppBusinessProcessResult result = new AppBusinessProcessResult()
+            {
+                OutputData = null,
+                NumberSuccessfullyProcessed = 0,
+                NumberOfExceptions = 0
+            };
+
+            faceClient = new FaceClient(
+                new ApiKeyServiceClientCredentials(subscriptionKey),
+                new System.Net.Http.DelegatingHandler[] { })
+            {
+                Endpoint = faceEndpoint
+            };
+
+            using (SqlConnection conn = RequestContext.OpenAppDBConnection())
+            {
+                List<Task> tasks = new List<Task>();
+                Guid idSetRegisterID = Guid.Empty;
+                SPWrapConstituentPhotoAnalysis.USR_USP_CONSTITUENTPHOTOANALYSISPROCESS_GETPARAMETERS.ResultRow parameterRow = SPWrapConstituentPhotoAnalysis.USR_USP_CONSTITUENTPHOTOANALYSISPROCESS_GETPARAMETERS.WrapperRoutines.ExecuteRow(conn, ProcessContext.ParameterSetID);
+                if (parameterRow != null && (parameterRow.IDSETREGISTERID != Guid.Empty))
+                {
+                    idSetRegisterID = parameterRow.IDSETREGISTERID;
+                }
+                else
+                {
+                    throw new System.Exception("Could not retrieve parameters.");
```

Comment:
Is this the right exception type? I thought we usually use ServiceException... or is that only in UIModel event handlers?

Reply:
Changed.